include/data/nodeid.h

```diff
@@ -54,5 +54,9 @@ class NodeID {
   explicit operator bool() const;
 };
 
+struct NodeIDHash {
```

Comment:
Hmm, any reason why not provide specialization for std::hash instead? If we did that I think we wouldn't need to explicitly provide it as parameter to std unordered containers.